### PR TITLE
implement tagging feature for patients

### PR DESCRIPTION
**Pull request by kennethszj**
#### **Changes:**
1. **Add Patient with Tag:**
   - Users can now add patients along with an optional tag using the `add` command:
     ```
     add <name> /tag <tag>
     ```
   - Example: `add John t/smoker` creates a patient named John with the tag `[smoker]`.

2. **Edit Patient Tag:**
   - The `edit` command allows users to modify the tag of an existing patient based on their index in the list:
     ```
     edit <index> /tag <new_tag>
     ```
   - Example: `edit 1 /tag high priority` updates the tag for the first patient to `[high priority]`.

   - Users can clear a patient’s tag by entering `t/` without any value:
     ```
     edit <index> /tag
     ```
     - Example: `edit 2 /tag` clears the tag for the second patient.

3. **Tag Display:**
   - Patient tags are displayed in square brackets after the patient’s name when listing patients or selecting a specific patient.

#### **State Management:**
- Tagging functionality is only available in `MAIN_STATE`. Attempts to use the `edit` command in other states will return an error message.

### CODE REVIEW

File: src/main/java/seedu/duke/commands/EditPatientCommand.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package seedu.duke.commands;
```
Comment (AjayShanker-geek) [inline src/main/java/seedu/duke/commands/EditPatientCommand.java:1]: Is this just editing patient tag only? If yes, need a better name or this to edit all edit patient parameter in the future?

File: src/main/java/seedu/duke/ui/Ui.java [hunk 2]
```diff
@@ -22,9 +22,13 @@ public void showMainScreen() {
...
 
-    public void showTaskScreen(String patientName) {
+    public void showTaskScreen(String patientName, String patientTag) {
+        String tagFormatted = "";
+        if (patientTag != null) {
+            tagFormatted = " [" + patientTag + "]";
+        }
         showLine();
-        System.out.println("Patient: " + patientName);
+        System.out.println("Patient: " + patientName + tagFormatted);
```
Comment (AjayShanker-geek) [inline src/main/java/seedu/duke/ui/Ui.java]: How about passing in Patient object? This is allow to get patient `name` and `getFormattedTag`

File: src/main/java/seedu/duke/commands/SelectPatientCommand.java [hunk 3]
```diff
@@ -34,8 +34,14 @@ public CommandResult execute() {
 
         try {
             String patientName = hospital.getPatient(index).getName();
-            String resultMessage = String.format(MESSAGE_SUCCESS, patientName);
+            String patientTag = hospital.getPatient(index).getTag();
+            String formattedTag = "";
+            if (patientTag != null){
+                formattedTag = " [" + patientTag + "]";
+            }
+            String resultMessage = String.format(MESSAGE_SUCCESS, patientName + formattedTag);
```
Comment (AjayShanker-geek) [inline src/main/java/seedu/duke/commands/SelectPatientCommand.java]: Same, you can make use of the patient object. 

```
Patient p = hospital.getPatient(index)

String resultMessage = String.format(MESSAGE_SUCCESS, p.name + p.getFormattedTag);
```

File: src/main/java/seedu/duke/parser/AddParser.java [hunk 4]
```diff
@@ -9,8 +9,14 @@ public class AddParser implements CommandParser{
     @Override
     public Command execute(String line, State state) {
         if (state.getState() == StateType.MAIN_STATE) {
-            String patientName = line.substring("add ".length());
-            return new AddPatientCommand(patientName);
+            String[] parts = line.split(" t/");
```
Comment (AjayShanker-geek) [inline src/main/java/seedu/duke/parser/AddParser.java]: This is the format `/tag (optional)`

File: src/main/java/seedu/duke/commands/EditPatientCommand.java (hunk 1 above)
Comment (kennethszj) [reply src/main/java/seedu/duke/commands/EditPatientCommand.java:1]: For now is just to edit tags. However in the addressbook3, I see that they have multiple parameters that we don't have yet. So maybe we need to discuss this further to come to a conclusion.

File: src/main/java/seedu/duke/parser/AddParser.java (hunk 4 above)
Comment (kennethszj) [reply src/main/java/seedu/duke/parser/AddParser.java]: will change it to the correct format

File: src/main/java/seedu/duke/ui/Ui.java (hunk 2 above)
Comment (kennethszj) [reply src/main/java/seedu/duke/ui/Ui.java]: good idea. will change it to the suggested implementation

File: src/main/java/seedu/duke/commands/SelectPatientCommand.java (hunk 3 above)
Comment (kennethszj) [reply src/main/java/seedu/duke/commands/SelectPatientCommand.java]: good idea
will change it to the suggested implementation

**Review by lowjunchen — APPROVED** [review]
LGTM, I will integrate the code into Parser to check if it works

**Pull request merged**